### PR TITLE
fix(api): filter dq by managing org

### PR DESCRIPTION
**Pull request by Orta21**
Ref. metriport/metriport#1040

Ticket: #1040 

### Dependencies

- Upstream: none
- Downstream: none

### Description

Update internal endpoint to add a flag to query for only a specific managing orgs endpoints

### Testing

_[Regular PRs:]_

- Local
  - [x] Branched to staging
- Staging
  - [ ] Queries right endpoint
- Production
  - [ ] Queries right endpoint

Check each PR.

### Release Plan

- [ ] Merge this

### CODE REVIEW

File: packages/api/src/command/medical/document/document-query.ts [hunk 1]
```diff
@@ -49,6 +49,7 @@ export async function queryDocumentsAcrossHIEs({
   forceQuery = false,
   forceCommonwell = false,
   forceCarequality = false,
+  cqManagingOrgFilter,
```
Comment (leite08) [inline packages/api/src/command/medical/document/document-query.ts:52]: Should we indicate whether this is an ID or name? Like `cqManagingOrgName`

Comment (Orta21) [reply packages/api/src/command/medical/document/document-query.ts:52]: Its a name - ill update

File: packages/api/src/external/carequality/document/filter-oids-by-managing-org.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  });
+
+  if (!managingOrg) {
+    return [];
+  }
+
+  const managingOrgChildren = await CQDirectoryEntryModel.findAll({
+    where: {
+      managingOrganizationId: {
+        [Op.like]: "%" + managingOrg.id + "%",
```
Comment (leite08) [inline packages/api/src/external/carequality/document/filter-oids-by-managing-org.ts:20]: Do we need wildcard at the start as well? It prevents the usage of indexes

File: packages/api/src/external/carequality/document/query-documents.ts [hunk 3]
```diff
@@ -107,12 +110,17 @@ export async function getDocumentsFromCQ({
 
     const initiator = await getCqInitiator(patient);
 
+    let cqLinks = linksWithDqUrl;
+    if (cqManagingOrgFilter) {
+      cqLinks = await filterCqLinksByManagingOrg(cqManagingOrgFilter, linksWithDqUrl);
+    }
```
Comment (leite08) [inline packages/api/src/external/carequality/document/query-documents.ts]: Not a huge deal, but we can avoid using variable and use constant by just using a simple ternary.

Constants are are our friends - they're predictable and help minimize bugs. 😄

File: packages/api/src/routes/medical/internal-docs.ts [hunk 4]
```diff
@@ -345,6 +345,7 @@ router.get(
  * @param req.query.patientId - The customer/account's ID.
  * @param req.query.facilityId - Optional; The facility providing NPI for the document query.
  * @param req.body Optional metadata to be sent through webhook. {"disableWHFlag": "true"} can be sent here to disable webhook.
+ * @param req.query.cqManagingOrgFilter - Optional; The CQ managing organization.
```
Comment (leite08) [inline packages/api/src/routes/medical/internal-docs.ts]: nit: "... name."

**Review by Goncharo — APPROVED** [review]

**Review by leite08 — APPROVED** [review]

**Pull request merged**